### DISCUSSION TITLE
Can I add additional `select` on my query? (when QueryClient has `select`)

### DISCUSSION THREAD

**Discussion by buglessir** [TanStack/query#3146] [General] [answered]
I've used `select` once in `QueryClient` to change the data:

`(we are using the axios and it returns all response and I should extract the data)`

```
const appQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      select: (data) => data.data
    },
  },
});
```

but I'm using a custom GridView and I should send this data model to that component:

```
{
  currentData: data.rows,
  total: data.total,
  page: data.page
}
```

I was thinking to use another `select` for that specific query to transform data... but I can't keep both!
I mean, if I add `select` to my query that will override client's `select` and query miss the `data.data`
Do you have any idea? should I use another state to change data and send it to the Grid? or there is an option or config to keep both?

**Comment by TkDodo** [comment, marked as answer]
there is only one select, no chaining. I would just make a simple transformation like that in the custom hook:

```
const useMyQuery = () => {
  const query = useQuery(...)

  return {
    ...query,
    data: React.useMemo(() => ({
      currentData: query.data.rows,
      total: query.data.total,
      page: query.data.page
    }), [query.data])
  }
}
```
